src/Middleware/CasProxyTicketSessionMiddleware.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace OpenEuropa\EPoetry\Middleware;
+
+use Http\Promise\Promise;
+use Phpro\SoapClient\Middleware\Middleware;
+use Phpro\SoapClient\Middleware\MiddlewareInterface;
+use Psr\Http\Message\RequestInterface;
+use Symfony\Component\HttpFoundation\Session\Session;
+
+class CasProxyTicketSessionMiddleware extends Middleware implements MiddlewareInterface
+{
+    private const PGT_ATTRIBUTE = 'cas_pgt';
+
+    /**
+     * The user session.
+     *
+     * @var Symfony\Component\HttpFoundation\Session\Session
+     */
+    protected $session;
+
+    /**
+     * CasProxyTicketMiddleware constructor.
+     *
+     * @param session $session
+     *   The user session
+     */
+    public function __construct(Session $session)
```

Comment:
Shouldn't we use PHP's `\SessionHandlerInterface` here? This will allow us to remove the `"symfony/http-foundation": "^4.2"` dependency. When integrating it with Drupal we can simply pass the `session_handler` service:

```yml
  session_handler:
    alias: session_handler.storage
  session_handler.storage:
    class: Drupal\Core\Session\SessionHandler
    arguments: ['@request_stack', '@database']
    tags:
      - { name: backend_overridable }
```

That service's class `Drupal\Core\Session\SessionHandler` implements `\SessionHandlerInterface`.

We still need to make sure we can access the proxy ticket value from there, of that I'm not entirely sure but I would expect we could.

Reply:
`Drupal\Core\Session\SessionHandler` handles sessions directly from DB. It's not what we need here.
Cas module uses `\Symfony\Component\HttpFoundation\Session\SessionInterface`.

I've now updated the `CasProxyTicketSessionMiddleware` class to use this interface (instead of this class that implements it: `\Symfony\Component\HttpFoundation\Session\Session`).